aten/src/ATen/LegacyTHFunctionsCUDA.h

```diff
@@ -28,6 +28,8 @@ std::tuple<Tensor &,Tensor &> _th_mode_out(Tensor & values, Tensor & indices, co
 std::tuple<Tensor,Tensor> _th_mode(const Tensor & self, int64_t dim, bool keepdim);
 std::tuple<Tensor &,Tensor &> _th_sort_out(Tensor & values, Tensor & indices, const Tensor & self, int64_t dim, bool descending);
 std::tuple<Tensor,Tensor> _th_sort(const Tensor & self, int64_t dim, bool descending);
+std::tuple<Tensor &,Tensor &> _th_sort_out_stable(Tensor & values, Tensor & indices, const Tensor & self, c10::optional<bool> stable, int64_t dim, bool descending);
```

Comment:
We shouldn't be adding new function to TH! Can we put this in ATen?

Reply:
I decided to add it in order to indicate it's "hacky" nature. Wouldn't it be confusing to have in ATen while not being ported? Would it be possible to call ATen from TH? The legacy functions will need to call it.

Reply:
In fact, porting `sort` for CUDA is on my TODO list, so I could do it there.

Reply:
Sounds like this is already tracked, so no worries for now.